source/GameData.cpp

```diff
@@ -722,6 +722,23 @@ const vector<string> &GameData::Category(const CategoryType type)
 
 
 
+const std::pair<bool, std::pair<bool, bool>> GameData::BunkType(const std::string type)
```

Comment:
A class or a struct

Reply:
It could be a struct, but then you can't really put logic in it (not like outfit.cpp is huge, could be fine, but conceptually not all outfits have bunks)
However,
If you make it a class, all the logic you placed in outfit.cpp could be moved there, and all could be dealt with internally; not all outfits have crew attributes. Also, said class could be used at other points, like in cargohold.cpp, which I think you need to modify for this to work with missions properly.
Said class would be held contained properly in UniverseObject, and worry about it's own things.

Reply:
Just to get this right. You want me to add a class in universe objects and this class gets used in every outfit?
Excuse me, I am a little bit confused.

Reply:
I will try something and you tell me if I did it right.

Reply:
So I implemented the class, but I believe you still want some changes. Please specify them, so I can implement them asap